pkg/controlplane/client.go

```diff
@@ -113,11 +114,20 @@ func (c *client) IsActive() bool {
 
 // setActive the peer status (active or not).
 func (c *client) setActive(active bool) {
+	var activePrevState bool
 	c.lock.Lock()
-	defer c.lock.Unlock()
-	c.active = active
-	if active || c.lastSeen.IsZero() {
-		c.lastSeen = time.Now()
+	{
+		activePrevState = c.active
+		c.active = active
+		if active || c.lastSeen.IsZero() {
+			c.lastSeen = time.Now()
+		}
+	}
```

Comment:
```suggestion
	activePrevState := c.active
	c.active = active
	if active || c.lastSeen.IsZero() {
		c.lastSeen = time.Now()
	}
```

And remove `var activePrevState bool`